source/bases/Win32Service.c

```diff
@@ -291,19 +301,25 @@ static int Service_SetupPython(udt_ServiceInfo *info)
 
     // acquire the __main__ module
     module = PyImport_ImportModule("__main__");
-    if (!module)
-        return LogPythonException("unable to import __main__");
+    if (!module) {
+        error_message = "unable to import __main__";
+        goto end;
```

Comment:
I am not at all a fan of goto. If necessary, you can create a separate method (Service_SetupPythonHelper or some such) which will perform the work. There is no real need, though, since if this fails, the entire executable will fail and the process will end.

Reply:
I don't like 'goto' either. I recently added a new function in util and used 'if succeded' but it's not all good either. As I have been reading some python codes and I see that in some situations they surrender to goto, I surrendered too....